### DISCUSSION TITLE
CSV import: not able to use properly dateformat

### DISCUSSION THREAD

**Discussion by aborruso** [duckdb/duckdb#10951] [Q&A] [answered]
Hi,
I'm sure I'm missing something trivial, but I can't figure out what.

I have this CSV input file: https://gist.github.com/aborruso/668b74b13a95c159ff59e789a6289abd

In the last row I have in two columns this date value: `23-MAY-2011`.

If I run

```
duckdb -c "select * from read_csv_auto('tmp.csv',auto_type_candidates=['BIGINT'],parallel=FALSE,dateformat='%d-%b-%y',types={'DATA_ULTIMA_MODIFICA_UTENTE':'DATE','DATA_CHIUSURA_REVOCA':'DATE','DATA_GENERAZIONE_CUP':'DATE'})"
```

I have the below error. And I don't understand why.
`%d-%b-%y` seems to me the proper format for `23-MAY-2011`.

What am I not understanding? What am I doing wrong?

Thank you

```
Error: Conversion Error: CSV Error on Line: 2
Error when converting column "DATA_ULTIMA_MODIFICA_UTENTE".
Could not parse string "23-MAY-2011" according to format specifier "%d-%b-%y"
23-MAY-2011
         ^
Error: Full specifier did not match: trailing characters

  file=tmp.csv
  delimiter = , (Auto-Detected)
  quote = " (Auto-Detected)
  escape = " (Auto-Detected)
  new_line = \n (Auto-Detected)
  header = true (Auto-Detected)
  skip_rows = 0 (Auto-Detected)
  date_format = %d-%b-%y (Set By User)
  timestamp_format =  (Auto-Detected)
  null_padding=0
  sample_size=20480
  ignore_errors=0
  all_varchar=0
CSV Sniffer: Sniffer detected value different than the user input for the Date Format options
 Set: %d-%b-%y Sniffed:
```

**Comment by Tishj** [comment, marked as answer]
Have a look at <https://duckdb.org/docs/data/csv/overview>
There is a `dateformat` option, with a link to a page explaining all the details of the dateformat

It looks to me like your last format should be `%Y` instead of `%y`

**Reply by aborruso** [reply]
> It looks to me like your last format should be %Y instead of %y

I feel very stupid. Thank you very much
